internal/project/common/repository.go

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/google/go-github/v32/github"
+
+	"github.com/talos-systems/kres/internal/dag"
+	"github.com/talos-systems/kres/internal/project/meta"
+)
+
+// Repository sets up repository settings.
+type Repository struct {
+	dag.BaseNode
+
+	meta *meta.Options
+
+	MainBranch     string   `yaml:"mainBranch"`
+	EnforeContexts []string `yaml:"enforceContexts"`
+}
+
+// NewRepository initializes Repository.
+func NewRepository(meta *meta.Options) *Repository {
+	return &Repository{
+		BaseNode: dag.NewBaseNode("repository"),
+
+		meta: meta,
+
+		MainBranch: "master",
+		EnforeContexts: []string{
+			"continuous-integration/drone/pr",
+			"status-lgtm",
+		},
+	}
+}
+
+// CompileGitHub implements github.Compiler.
+func (r *Repository) CompileGitHub(client *github.Client) error {
+	branchProtection, resp, err := client.Repositories.GetBranchProtection(context.Background(), r.meta.GitHubOrganization, r.meta.GitHubRepository, r.MainBranch)
+	if err != nil {
+		if resp.StatusCode != http.StatusNotFound {
+			return err
+		}
+	}
+
+	req := github.ProtectionRequest{
+		RequireLinearHistory: github.Bool(true),
```

Comment:
this is the most important part: do we all agree that it's the settings we want?

Reply:
Plus one from me. Linear histories are awesome :smile: